ci(renovate): EF Core v10 PRs (#2379, #2380) bump the net8.0-pinned entry and always fail with NU1202 — add matchCurrentVersion rule

## Summary

Renovate keeps opening EF Core v10 PRs (#2379, #2380) that can never pass CI. Their CI runs fail with:

```
tests/Dekaf.Tests.Unit/Dekaf.Tests.Unit.csproj : error NU1202: Package Microsoft.EntityFrameworkCore.Sqlite 10.0.10 is not compatible with net8.0 (.NETCoreApp,Version=v8.0). Package Microsoft.EntityFrameworkCore.Sqlite 10.0.10 supports: net10.0
```

## Root cause

`Directory.Packages.props` already handles EF's per-TFM shipping correctly with two conditioned entries:

```xml
<!-- Entity Framework Core ships per-TFM: the 8.x train is the last supporting net8.0. -->
<ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
  <PackageVersion Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.29" />
  ...
<ItemGroup Condition="'$(TargetFramework)' != 'net8.0'">
  <PackageVersion Include="Microsoft.EntityFrameworkCore.Sqlite" Version="10.0.10" />
```

Renovate doesn't understand the MSBuild condition — the PR diff for #2380 bumps the **net8.0-conditioned** `8.0.29` entry to `10.0.10`, which is exactly the combination the comment above it says cannot work. The PRs will re-open/rebase forever and burn CI on every push.

## Fix

Add a Renovate `packageRules` entry that keeps the 8.x train on 8.x while leaving the v10 entry free to update (`matchCurrentVersion` applies per dependency instance, so it distinguishes the two entries of the same package):

```json
{
  "matchPackageNames": ["/^Microsoft\.EntityFrameworkCore\./"],
  "matchCurrentVersion": "/^8\./",
  "allowedVersions": "<9",
  "description": "EF 8.x train pinned for the net8.0 TFM (Directory.Packages.props conditioned entry); 9+/10+ do not support net8.0"
}
```

Then close #2379 and #2380 (they become permanently unwanted). The v10-conditioned entries continue receiving normal 10.x patch updates (e.g. #2381-style bumps).

The alternative — dropping the net8.0 target from the test projects — is a separate, larger decision and not required to stop the noise.

--- a/renovate.json
+++ b/renovate.json
@@ -1,3 +1,13 @@
 {
-  "$schema": "https://docs.renovatebot.com/renovate-schema.json"
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "packageRules": [
+    {
+      "description": "EF 8.x train pinned for the net8.0 TFM (Directory.Packages.props conditioned entry); 9+/10+ do not support net8.0",
+      "matchPackageNames": [
+        "/^Microsoft\\.EntityFrameworkCore\\./"
+      ],
+      "matchCurrentVersion": "/^8\\./",
+      "allowedVersions": "<9"
+    }
+  ]
 }

